test/e2e/utils.go

```diff
@@ -40,39 +42,70 @@ var (
 	bondConunter  = 0
 )
 
-func writePodsLogs(namespace string, sinceTime time.Time, writer io.Writer) error {
+func writePodsLogs(namespace string, sinceTime time.Time, testFailed bool) error {
 	if framework.Global.LocalOperator {
 		return nil
 	}
+	OpenTestLogFile("pod", func(writer io.ReadWriteCloser){
+		podLogOpts := corev1.PodLogOptions{}
+		podLogOpts.SinceTime = &metav1.Time{sinceTime}
+		podList := &corev1.PodList{}
+		err := framework.Global.Client.List(context.TODO(), podList, &dynclient.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		podsClientset := framework.Global.KubeClient.CoreV1().Pods(namespace)
+
+		for _, pod := range podList.Items {
+			appLabel, hasAppLabel := pod.Labels["app"]
+			if !hasAppLabel || appLabel != "kubernetes-nmstate" {
+				continue
+			}
+			req := podsClientset.GetLogs(pod.Name, &podLogOpts)
+			podLogs, err := req.Stream()
+			if err != nil {
+				io.WriteString(GinkgoWriter, fmt.Sprintf("error in opening stream: %v\n", err))
+				continue
+			}
+			defer podLogs.Close()
+			rawLogs, err := ioutil.ReadAll(podLogs)
+			if err != nil {
+				io.WriteString(GinkgoWriter, fmt.Sprintf("error reading kubernetes-nmstate logs: %v\n", err))
+				continue
+			}
+			formattedLogs := strings.Replace(string(rawLogs), "\\n", "\n", -1)
+			io.WriteString(writer, formattedLogs)
+			// If test failed print pods logs to screen
+			if testFailed{
+				io.WriteString(GinkgoWriter, formattedLogs)
+			}
+		}
+	})
+	return nil
+}
 
-	podLogOpts := corev1.PodLogOptions{}
-	podLogOpts.SinceTime = &metav1.Time{sinceTime}
-	podList := &corev1.PodList{}
-	err := framework.Global.Client.List(context.TODO(), podList, &dynclient.ListOptions{})
+func printDeviceStatus(node string) {
+	output, err := runAtNode(node, "nmcli", "c", "s")
 	Expect(err).ToNot(HaveOccurred())
-	podsClientset := framework.Global.KubeClient.CoreV1().Pods(namespace)
+	GinkgoWriter.Write([]byte(fmt.Sprintf("\n***** Connection status on node %s *****\n\n %s", node, output)))
+	GinkgoWriter.Write([]byte(fmt.Sprintf("\n***** Done Connection status on node %s*****\n", node)))
 
-	for _, pod := range podList.Items {
-		appLabel, hasAppLabel := pod.Labels["app"]
-		if !hasAppLabel || appLabel != "kubernetes-nmstate" {
-			continue
-		}
-		req := podsClientset.GetLogs(pod.Name, &podLogOpts)
-		podLogs, err := req.Stream()
-		if err != nil {
-			io.WriteString(writer, fmt.Sprintf("error in opening stream: %v\n", err))
-			continue
-		}
-		defer podLogs.Close()
-		rawLogs, err := ioutil.ReadAll(podLogs)
-		if err != nil {
-			io.WriteString(writer, fmt.Sprintf("error reading kubernetes-nmstate logs: %v\n", err))
-			continue
-		}
-		formattedLogs := strings.Replace(string(rawLogs), "\\n", "\n", -1)
-		io.WriteString(writer, formattedLogs)
-	}
-	return nil
+	output, err = runAtNode(node, "nmcli", "d", "s")
```

Comment:
full path?

Reply:
why is a full path better? Depending on a proper definition of PATH is not bad imho. I was once convinced that full paths are somewhat safer (you are not exposed to someone adding his own ~/bin dir into your PATH) but I don't consider it a real threat here.

Reply:
will change just for matter of consistency in the same function :)